js/articleView.js

```diff
@@ -1,75 +1,84 @@
-  $('.article-body *:nth-of-type(n+2)').hide();
-  $('article').on('click', 'a.read-on', function(e) {
-    e.preventDefault();
-    if($(this).text() === 'Read on →') {
-      $(this).parent().find('*').fadeIn();
-      $(this).html('Show Less &larr;');
-    } else {
-      $('body').animate({
-        scrollTop: ($(this).parent().offset().top)
-      },200);
-      $(this).html('Read on &rarr;');
-      $(this).parent().find('.article-body *:nth-of-type(n+2)').hide();
-    }
-  });
-};
+  articleView.handleMainNav = function () {
+    console.log('into handleMainNav');
+    $('.main-nav').on('click', '.tab', function(e) {
+      $('.tab-content').hide();
+      $('#' + $(this).data('content')).fadeIn();
+    });
+    $('.main-nav .tab:first').click();
+  };
 
-articleView.renderIndexPage = function() {
-  Article.allArticles.forEach(function(a){
-    $('#articles').append(a.toHtml('#article-template'));
-    if($('#category-filter option:contains("'+ a.category + '")').length === 0) {
-      $('#category-filter').append(a.toHtml('#category-filter-template'));
-    };
-    if($('#author-filter option:contains("'+ a.author + '")').length === 0) {
-      $('#author-filter').append(a.toHtml('#author-filter-template'));
-    };
-  });
-  articleView.handleCategoryFilter();
-  articleView.handleAuthorFilter();
-  articleView.handleMainNav();
-  articleView.setTeasers();
-};
+  articleView.setTeasers = function() {
+    $('h2').prev('p').remove();
+    $('h2').next('p').remove();
+    $('.article-body *:nth-of-type(n+2)').hide();
+    $('article').on('click', 'a.read-on', function(e) {
+      e.preventDefault();
+      if($(this).text() === 'Read on →') {
+        $(this).parent().find('*').fadeIn();
+        $(this).html('Show Less &larr;');
+      } else {
+        $('body').animate({
+          scrollTop: ($(this).parent().offset().top)
+        },200);
+        $(this).html('Read on &rarr;');
+        $(this).parent().find('.article-body *:nth-of-type(n+2)').hide();
+      }
+    });
+  };
 
-// TODO: DONE! invoke the retrieval process for our data!
-Article.fetchAll();
+  articleView.renderIndexPage = function() {
+    Article.allArticles.forEach(function(a){
+      //$('#articles').append(a.toHtml('#article-template'));
+      $('#articles').append(a.toHtml($('#article-template')));
+      if($('#category-filter option:contains("'+ a.category + '")').length === 0) {
+        $('#category-filter').append(a.toHtml('#category-filter-template'));
+      };
+      if($('#author-filter option:contains("'+ a.author + '")').length === 0) {
+        $('#author-filter').append(a.toHtml('#author-filter-template'));
+      };
+
+    });
+    articleView.handleCategoryFilter();
+    articleView.handleAuthorFilter();
+    articleView.handleMainNav();
+    articleView.setTeasers();
+  };
+
+// Invoke the retrieval process for our data!
+  Article.fetchAll(articleView.renderIndexPage);
+
+  module.articleView = articleView;
+})(window);
```

Comment:
Overall, very good syntax and clean code. Your spacing and indentation make the code easy to read. Your comments are very helpful as well. I think you could add more comments as well. I would like to do the same thing to my code. It makes it easier for diagnosing bugs and understanding what is happening in each area.